api/python/provisioner/commands/configure_setup.py

```diff
@@ -90,199 +98,166 @@ def __attrs_post_init__(self):
+                'netmask', 'public_ip', 'private_ip',
+                'public_interfaces', 'private_interfaces',
+                'id', 'roles', 'devices', 'interfaces'
+            ]
+
+            if len(val) > 1 and val[-1] in conf_values:
+                params[f'{val[-2]}_{val[-1]}'] = param_data[key]
+
             else:
-                logger.debug(f"Params generated: {params}")
-                params[val[-1]] = input[key]
+                params[val[-1]] = param_data[key]
+
+        logger.debug(f"Params generated: {params}")
         return params
 
     def _validate_params(self, input_type, content):
         params = self._parse_params(content)
         self.validate_map[input_type](**params)
 
-    def _parse_input(self, input):
-        for key in input:
-            if input[key] and "," in input[key]:
-                value = [f'\"{x.strip()}\"' for x in input[key].split(",")]
+    @staticmethod
+    def _parse_input(input_data):
+        for key in input_data:
+            keys_of_type_list = [
+                'interfaces', 'devices', 'roles'
+            ]
+            if input_data[key] and "," in input_data[key]:
+                value = [f'\"{x.strip()}\"' for x in input_data[key].split(",")]
                 value = ','.join(value)
-                input[key] = f'[{value}]'
-            elif 'interfaces' in key or 'roles' in key:
-                # special case single value as array
-                # Need to fix this array having single value
-                input[key] = f'[\"{input[key]}\"]'
+                input_data[key] = f'[{value}]'
+
+            elif key in keys_of_type_list:
+                input_data[key] = f'[\"{input_data[key]}\"]'
+
             else:
-                if input[key]:
-                    if input[key] == 'None':
-                        input[key] = '\"\"'
+                if input_data[key]:
+                    if input_data[key] == 'None':
+                        input_data[key] = '\"\"'
                     else:
-                        input[key] = f'\"{input[key]}\"'
+                        input_data[key] = f'\"{input_data[key]}\"'
                 else:
-                    input[key] = UNCHANGED
+                    input_data[key] = UNCHANGED
 
-    def _parse_pillar_key(self, key):
+    @staticmethod
+    def _parse_pillar_key(key):
         pillar_key = deepcopy(key)
         return pillar_key.replace(".", "/")
 
+    def _update_pillar_data(self, content, section, pillar_type):
+        """
+        Set pillar data.
+
+        Config content of default sections will be applied
+        across all nodes.
+
+        """
+        for pillar_key in content[section]:
+
+            key = f'{pillar_type}/{self._parse_pillar_key(pillar_key)}'
+            value_to_set = f"{content[section][pillar_key]}"
+
+            run_subprocess_cmd([
+                   "provisioner", "pillar_set",
+                   key, value_to_set])
+
+        if content.get('cluster', None):
```

Comment:
that block of code would be run each time that function is called, looks not clean enough

Reply:
Done. Moved it out of iteration.